working/2364 - primary constructors/feature-specification.md

```diff
@@ -424,6 +474,13 @@ default value.
   removed from the parameter in _L2_, and it is added to the instance
   variable declaration named `p`.
 
+If there are any assertions following the formal parameter list _L_ then
+_k_ has an initializer list with the same assertions in the same order. The
+expressions in the assertions are subject to a transformation similar to
+the one which is used with default values, such that the meaning of the
+expression is the same when it occurs in the class header and when it
+occurs in the initializer list of _k_.
```

Comment:
Nice hand-wave, but *how* is that done when the expressions can refer to parameters and type parameters.

Reply:
It's quite similar to the situation with mixins: We claim that a mixin application will give rise to a fresh, implicitly induced class, and the instance members of that class are "obtained" from the mixin declaration.

So we might want to specify that there is a way to obtain "the semantics" of a given member declaration such that we can say that we "add that semantics" to a class.

We could also specify that an AST can be resolved, and we can freely move a resolved AST around because none of the identifiers in that AST will be looked up again.

Reply:
I've rewritten this paragraph to be more explicit on the required outcome, rather than specifying a particular desugaring strategy.